src/main/java/org/littleshoot/proxy/impl/ClientToProxyConnection.java

```diff
@@ -321,6 +338,16 @@ private ConnectionState doReadHTTPInitial(HttpRequest httpRequest) {
         }
     }
 
+    private boolean isProxyRequested() {
+        // HTTPS requests have uri without http scheme too
+        if (currentRequest.getMethod() == HttpMethod.CONNECT || sslEngine != null) {
```

Comment:
Does this mean that HTTPS requests to the proxy will not be caught by this logic?